adapters/repos/db/lsmkv/segment_group_compaction.go

```diff
@@ -262,6 +262,30 @@ func (sg *SegmentGroup) replaceCompactedSegments(old1, old2 int,
 		sg.segments[old2].countNetAdditions
 	sg.maintenanceLock.RUnlock()
 
+	sg.maintenanceLock.Lock()
+
+	leftSegment := sg.segments[old1]
+	rightSegment := sg.segments[old2]
+	newSegmentId := "segment-" + segmentID(leftSegment.path) + "_" + segmentID(rightSegment.path)
+
+	// delete any existing bloom tmp files in the form of segment-<seg1>_<seg2>*bloom.tmp
+	tmpFiles, err := filepath.Glob(filepath.Join(sg.dir, newSegmentId+"*bloom.tmp"))
+	if err != nil {
+		sg.maintenanceLock.Unlock()
+		return errors.Wrap(err, "glob tmp files")
+	}
+
+	for _, tmpFile := range tmpFiles {
+		err = os.Remove(tmpFile)
+		if err != nil {
+			sg.maintenanceLock.Unlock()
+			return errors.Wrap(err, "remove tmp file")
+		}
+	}
+
+	sg.maintenanceLock.Unlock()
```

Comment:
hmm, just a suggestion, maybe running this code within an anonymous function would simplify the locking / unlocking logic?
```
func() {
  sg.maintenanceLock.Lock()
  defer sg.maintenanceLock.RUnlock()

  // all code goes here
}()
```

or maybe putting this operation to a separate method?